src/qml/views/timeline/timeline.qml

```diff
@@ -232,6 +235,103 @@ Rectangle {
+                                            ParentChange { target: dragItem; parent: trackHeaders.parent }
+                                            PropertyChanges { target: dragItem; anchors.fill: undefined }
+                                        },
+                                        State {
+                                            when: !dragMouseArea.drag.active
+                                            ParentChange { target: dragItem; parent: trackHead }
+                                            PropertyChanges { target: dragItem; anchors.fill: dragMouseArea }
+                                        }
+                                    ]
+                                }
+                            }
+
+                            DropArea {
+                                id: dropArea
+                                property var trackHead: trackHead
+                                anchors.fill: parent
+                                keys: ["trackHeader"]
+                                property bool containsValidDrag: false
+                                MessageDialog {
+                                    id: movePrompt
+                                    property var fromTrack
+                                    title: qsTr("Move track %1").arg(fromTrack.trackName)
+                                    icon: StandardIcon.Question
+                                    text: qsTr("Are you sure you want to move track %1?").arg(fromTrack.trackName)
+                                    standardButtons: StandardButton.Yes | StandardButton.No
+                                    onYes: {
+                                        timeline.moveTrack(fromTrack.trackIndex, trackHead.trackIndex)
+                                    }
+                                }
+                                onEntered: {
+                                    if (trackHead.isVideo == drag.source.trackHead.isVideo) {
+                                        containsValidDrag = true
+                                    } else {
+                                        containsValidDrag = false
+                                    }
+                                }
+                                onExited: {
+                                    containsValidDrag = false
+                                }
+                                onDropped: {
+                                    if (drop.proposedAction == Qt.MoveAction &&
+                                        trackHead.isVideo == drop.source.trackHead.isVideo &&
+                                        trackHead.trackIndex != drop.source.trackHead.trackIndex) {
+                                        drop.acceptProposedAction()
+                                        movePrompt.fromTrack = drop.source.trackHead
+                                        movePrompt.visible = true
+                                    }
```

Comment:
Overall, I like it and think its basically ready to merge. Only other comment is perhaps here when the drop is rejected we show a status message in the designated status message area.

Reply:
Added some status messages - good idea.